src/hypothesis/extra/datetime.py

```diff
@@ -15,111 +15,120 @@
 #
 # END HEADER
 
+"""This module provides time and date related strategies.
+
+It depends on the ``pytz`` package, which is stable enough that almost any
+version should be compatible - most updates are for the timezone database.
+
+"""
+
 from __future__ import division, print_function, absolute_import
 
 import datetime as dt
 
 import pytz
 
-import hypothesis.internal.conjecture.utils as cu
+import hypothesis.strategies as st
 from hypothesis.errors import InvalidArgument
-from hypothesis.strategies import defines_strategy
-from hypothesis.searchstrategy.strategies import SearchStrategy
-
-
-class DatetimeStrategy(SearchStrategy):
-
-    def __init__(self, allow_naive, timezones, min_year=None, max_year=None):
-        self.allow_naive = allow_naive
-        self.timezones = timezones
-        self.min_year = min_year or dt.MINYEAR
-        self.max_year = max_year or dt.MAXYEAR
-        for a in ['min_year', 'max_year']:
-            year = getattr(self, a)
-            if year < dt.MINYEAR:
-                raise InvalidArgument(u'%s out of range: %d < %d' % (
-                    a, year, dt.MINYEAR
-                ))
-            if year > dt.MAXYEAR:
-                raise InvalidArgument(u'%s out of range: %d > %d' % (
-                    a, year, dt.MAXYEAR
-                ))
-
-    def do_draw(self, data):
-        while True:
-            try:
-                result = dt.datetime(
-                    year=cu.centered_integer_range(
-                        data, self.min_year, self.max_year, 2000
-                    ),
-                    month=cu.integer_range(data, 1, 12),
-                    day=cu.integer_range(data, 1, 31),
-                    hour=cu.integer_range(data, 0, 24),
-                    minute=cu.integer_range(data, 0, 59),
-                    second=cu.integer_range(data, 0, 59),
-                    microsecond=cu.integer_range(data, 0, 999999)
-                )
-                if (
-                    not self.allow_naive or
-                    (self.timezones and cu.boolean(data))
-                ):
-                    result = cu.choice(data, self.timezones).localize(result)
-                return result
-
-            except (OverflowError, ValueError):
-                pass
-
-
-@defines_strategy
+from hypothesis._settings import note_deprecation
+
+__all__ = ['timezones', 'datetimes', 'dates', 'times']
+
+
+@st.cacheable
+@st.defines_strategy
+def timezones():
+    """Any timezone in the Olsen database, as a pytz tzinfo object.
+
+    This strategy minimises to UTC, or the smallest possible fixed
+    offset, and is designed for use with
+    :py:func:`hypothesis.strategies.datetimes`.
+
+    """
+    all_timezones = [pytz.timezone(tz) for tz in pytz.all_timezones]
+    static = [pytz.UTC] + sorted(
```

Comment:
Will do.  A StaticTzInfo represents a timezone that has always had a constant offset from UTC - no daylight savings, and no other tricks like moving across the date line.  IMO that makes them simpler than the others, and smaller absolute offsets simpler within that.